Add troubleshooting guide for Angular "First build never completed" error (#115)
oppia/oppia-web-developer-docs: Troubleshooting.md

    self.gen.throw(type, value, traceback)
  File "/home/user/opensource/oppia/scripts/servers.py", line 138, in managed_process
    raise Exception(
Exception: Process Angular Compiler(name="sh", pid=2350) exited unexpectedly with exit code 137

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/user/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/user/opensource/oppia/scripts/start.py", line 233, in <module>
    main()
  File "/home/user/opensource/oppia/scripts/start.py", line 229, in main
    dev_appserver.wait()
  File "/home/user/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 525, in __exit__
    raise exc_details[1]
  File "/home/user/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 510, in __exit__
    if cb(*exc_details):
  File "/home/user/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 382, in _exit_wrapper
    callback(*args, **kwds)

>   File "/home/pratiksolanki/opensource/oppia/scripts/start.py", line 132, in call_extend_index_yaml

Reviewer: ```suggestion
  File "/home/user/opensource/oppia/scripts/start.py", line 132, in call_extend_index_yaml
```

    extend_index_yaml.main()

>   File "/home/pratiksolanki/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main

Reviewer: ```suggestion
  File "/home/user/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main
```

>   File "/home/pratiksolanki/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main

Reviewer: ```suggestion
  File "/home/user/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main
```

Should we keep the logs anonymous?

Author: done (though note that these come from a public discussions post, so they're public already)

    with open(WEB_INF_INDEX_XML_PATH, 'r', encoding='utf-8') as f:

> FileNotFoundError: [Errno 2] No such file or directory: '/home/pratiksolanki/opensource/oppia/../cloud_datastore_emulator_cache/WEB-INF/appengine-generated/datastore-indexes-auto.xml'

Reviewer: ```suggestion
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/opensource/oppia/../cloud_datastore_emulator_cache/WEB-INF/appengine-generated/datastore-indexes-auto.xml'
```

```

> The error code 137 indicates that the Angular compiler ran out of memory. This happens because by default, WSL2 only gives the host operating system (Ubuntu in our case) 50% of the machine's RAM. If you have 8 GB RAM, for example, this means Ubuntu only gets 4 GB, which is not enough to run Oppia. To fix this, you need to follow [Microsoft's instructions](https://learn.microsoft.com/en-us/windows/wsl/wsl-config) to increase the amount of RAM given to the host operating system.

Reviewer: ```suggestion
The error code 137 indicates that the Angular compiler ran out of memory. This happens because WSL2 only gives the host operating system (Ubuntu, in our case) 50% of the machine's RAM by default. If you have 8 GB RAM, for example, Ubuntu only gets 4 GB, which is insufficient to run Oppia. To fix this, you need to follow [Microsoft's instructions](https://learn.microsoft.com/en-us/windows/wsl/wsl-config) to increase the amount of RAM given to the host operating system.
```

Author: done
